src/schema_registry/telemetry/middleware.py

```diff
@@ -20,12 +23,59 @@ async def telemetry_middleware(
     request: Request,
     call_next: Callable[[Request], Awaitable[Response]],
     tracer: Tracer = Provide[TelemetryContainer.tracer],
+    meter: Meter = Provide[TelemetryContainer.meter],
 ) -> Response:
     resource = request.url.path.split("/")[1]
     with tracer.get_tracer().start_as_current_span(name=f"{request.method}: /{resource}", kind=SpanKind.SERVER) as span:
+        span.add_event("Creating metering resources")
+        karapace_http_requests_in_progress: UpDownCounter = meter.get_meter().create_up_down_counter(
```

Comment:
Is this creating the counter or reusing the existing one? Similar question for the duration histogram and requests counter.

Reply:
Good point, I looked up the docs for [get_meter()](https://opentelemetry-python.readthedocs.io/en/latest/api/metrics.html#opentelemetry.metrics.MeterProvider.get_meter) and they also mention that it is uncertain if subsequent calls to this function return the same instrument.

I will move them out and define them concretely in another module, so we are not trying to access `get_meter()` on every API request.